src/audio/codec_adapter/codec_adapter.c

```diff
@@ -245,7 +245,7 @@ static int codec_adapter_prepare(struct comp_dev *dev)
 	}
 
 	/* Allocate local buffer */
-	buff_size = MAX(cd->period_bytes, codec->cpd.in_buff_size) * buff_periods;
+	buff_size = MAX(cd->period_bytes, codec->cpd.out_buff_size) * buff_periods;
```

Comment:
What type of data does the buffer actually hold? The input to the algorithm or the output from it? I hope this won't break when we eventually deal with the compress algorithms (compared to the decompress ones we're handling now).

Reply:
This buffer is used to copy the algorithm output buffer. Because usually we cannot consume all of it in one period we need to store it locally.

Reply:
Then it would probably better to call it `local_output_buffer` or something.

Reply:
@paulstelian97 I won't modify the existing buffer name. This is a good name choose by the initial commit of @mrajwa